EasyCommands/CommandParsers/ParameterProcessors.cs

```diff
@@ -116,1146 +319,307 @@ public static class ParameterProcessorRegistry {
                     List<ParameterProcessor> processors;
                     if (!parameterProcessorsByParameterType.TryGetValue(types[i].GetType(), out processors)) continue;
                     foreach (ParameterProcessor processor in processors) {
-                        if(!sortedParameterProcessors.Contains(processor)) InsertProcessor(processor, sortedParameterProcessors);
-                     }
+                        if (!sortedParameterProcessors.Contains(processor)) InsertProcessor(processor, sortedParameterProcessors);
+                    }
                 }
             }
 
             static void InsertProcessor(ParameterProcessor processor, List<ParameterProcessor> parameterProcessors) {
-                int priority = priorityByParameterProcessor[processor.GetType()];
+                int priority = processor.GetRank();
                 int i = 0;
-                while (i < parameterProcessors.Count() && priorityByParameterProcessor[parameterProcessors[i].GetType()] < priority) i++;
+                while (i < parameterProcessors.Count() && parameterProcessors[i].GetRank() < priority) i++;
                 parameterProcessors.Insert(i, processor);
             }
```

Comment:
Related to my other comment on taking advantage of rank living in `ParameterProcessor` and `IComparable`, I think you can get rid of this method and just add the appropriate processor to your list in `AddProcessors`. Then, at the end of your addition/searching logic, call `sort()` on it and it should be good to go. Should save a few more lines! (And should be a bit more efficient since you're not having to iterate over the list inside every iteration of `AddProcessors`).

Reply:
yep, i cleaned this up